fil-proofs-tooling/scripts/with-lock.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+# Inspired by http://mywiki.wooledge.org/BashFAQ/045
+
+failure_code=$1
+lockdir=$2
+shift 2
+
+if mkdir "$lockdir" > /dev/null 2>&1
+then
+    echo >&2 "successfully acquired lock (${lockdir})"
+
+    # Unlock (by removing dir) when the script finishes
+    trap 'echo >&2 "relinquishing lock (${lockdir})"; rm -rf "$lockdir"' EXIT
```

Comment:
I guess this is safe, as long as we know that `/tmp` will be cleared between runs on CircleCI (or wherever this script will run). If there's any chance of an orphaned lock dir, we should manually delete it before running the `retry.sh` commands or use a mechanism which cannot lead to that problem.

Reply:
> I guess this is safe, as long as we know that /tmp will be cleared between runs on CircleCI

This will definitely not be cleared between runs on CircleCI. The lock directory is on the Packet box, and by design it is not cleared between CircleCI runs (because it's used to coordinate across processes _on the Packet box).

Reply:
Ah… great.

Are you confident it *will* be cleared on restart of the Packet box if it goes down in the middle of a run? (If not, this seems like it will be a pain if/when it inevitably happens.)

Also, sorry for missing the point originally.